### PR TITLE
 #38: Adjusted First Provider mapping

### PR DESCRIPTION
**Pull request by oskarhinc**

### CODE REVIEW

File: src/main/java/org/benetech/servicenet/adapter/firstprovider/FirstProviderDataMapper.java [hunk 1]
```diff
@@ -48,11 +47,31 @@
     @Mapping(source = "eligibilityList", target = "eligibility")
     Eligibility extractEligibility(RawData rawData);
 
+    default PhysicalAddress extractPhysicalAddress(RawData rawData) {
+        String address2 = StringUtils.isNotBlank(rawData.getAddress2()) ? " " + rawData.getAddress2() : "";
```
Comment (pgesek) [inline src/main/java/org/benetech/servicenet/adapter/firstprovider/FirstProviderDataMapper.java]: add the space on line 55

Comment (pgesek) [reply src/main/java/org/benetech/servicenet/adapter/firstprovider/FirstProviderDataMapper.java]: If you want it conditional, you can StringUtils.join

File: src/main/java/org/benetech/servicenet/adapter/firstprovider/FirstProviderDataMapper.java [hunk 2]
```diff
@@ -48,11 +47,31 @@
...
+        return new PhysicalAddress()
+            .city(rawData.getCity())
+            .stateProvince(rawData.getStateProvince())
+            .address1(rawData.getAddress1() + address2);
+    }
+
+    default PostalAddress extractPostalAddress(RawData rawData) {
+        String address2 = StringUtils.isNotBlank(rawData.getAddress2()) ? " " + rawData.getAddress2() : "";
```
Comment (pgesek) [inline src/main/java/org/benetech/servicenet/adapter/firstprovider/FirstProviderDataMapper.java]: same as above

File: src/main/java/org/benetech/servicenet/adapter/firstprovider/FirstProviderDataMapper.java [hunk 3]
```diff
@@ -48,11 +47,31 @@
...
+            .stateProvince(rawData.getStateProvince())
+            .address1(rawData.getAddress1() + address2);
+    }
+
     default Service extractService(RawData rawData) {
-        return new Service().name(rawData.getServiceName()).url(rawData.getAppURL())
+        //TODO: extract fees from the description
+        //TODO: extract application procedure from the description
+        String additionalInfo = StringUtils.isNotBlank(rawData.getServiceAdditionalInfo())
+            ? " " + rawData.getServiceAdditionalInfo() : "";
```
Comment (pgesek) [inline src/main/java/org/benetech/servicenet/adapter/firstprovider/FirstProviderDataMapper.java]: same as above

File: src/main/java/org/benetech/servicenet/adapter/firstprovider/FirstProviderDataMapper.java [hunk 4]
```diff
@@ -65,6 +84,21 @@ default Service extractService(RawData rawData) {
         return Arrays.stream(langs).map(lang -> new Language().language(lang)).collect(Collectors.toSet());
     }
 
+    default Set<OpeningHours> extractOpeningHours(RawData rawData) {
+        String[] rawOpeningHours = rawData.getWeeklyOpenHoursRaw().split(", ");
+        Set<OpeningHours> result = new HashSet<>();
```
Comment (pgesek) [inline src/main/java/org/benetech/servicenet/adapter/firstprovider/FirstProviderDataMapper.java]: I think we might want to persist the order - any reason for using a set?

File: src/main/java/org/benetech/servicenet/adapter/firstprovider/FirstProviderDataMapper.java [hunk 5]
```diff
@@ -81,7 +115,9 @@ default Location extractLocation(RawData rawData) {
     }
 
     private String extractLocationName(RawData rawData) {
-        return LocationUtils.buildLocationName(rawData.getCity(), rawData.getStateProvince(), rawData.getAddress1());
+        String address2 = StringUtils.isNotBlank(rawData.getAddress2()) ? " " + rawData.getAddress2() : "";
```
Comment (pgesek) [inline src/main/java/org/benetech/servicenet/adapter/firstprovider/FirstProviderDataMapper.java]: same thing with putting space in the var - it''s missleading

File: src/main/java/org/benetech/servicenet/adapter/firstprovider/FirstProviderDataMapper.java (hunk 4 above)
Comment (oskarhinc) [reply src/main/java/org/benetech/servicenet/adapter/firstprovider/FirstProviderDataMapper.java]: Right

**Pull request merged**